src/buffer/out/OutputCellRect.cpp

```diff
@@ -21,14 +21,11 @@ OutputCellRect::OutputCellRect() noexcept :
 // Arguments:
 // - rows - Rows in the rectangle (height)
 // - cols - Columns in the rectangle (width)
-OutputCellRect::OutputCellRect(const size_t rows, const size_t cols) :
+OutputCellRect::OutputCellRect(const til::CoordType rows, const til::CoordType cols) :
     _rows(rows),
     _cols(cols)
 {
-    size_t totalCells;
-    THROW_IF_FAILED(SizeTMult(rows, cols, &totalCells));
-
-    _storage.resize(totalCells);
+    _storage.resize(gsl::narrow<size_t>(rows * cols));

```

Comment:
Semantic difference: `SizeTMult` throws if the multiplication would overflow; `narrow<size_t>` throws if the _result_ overflows (and this only works if the result of `rows * cols` fits inside a `til::CoordType`)

I think this means we're missing an overflow case where `rows*cols` overflows but the resulting value fits in a `size_t`.

```
14:14 <DHowett> { int32_t x = 2147483647; auto y = x; std::cout << (size_t)(y*x); }
14:14 <geordi> 1
```

Safe integer math is difficult.